resource_work_time_from_contracts/models/resource_calendar.py

```diff
@@ -0,0 +1,56 @@
+# SPDX-FileCopyrightText: 2024 Coop IT Easy SC
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+import datetime
+
+from odoo import models
+
+
+class ResourceCalendar(models.Model):
+    _inherit = "resource.calendar"
+
+    def get_first_last_attendance(self, date_from, date_to):
+        """
+        Get the first and last attendances between (and including) date_from
+        and date_to.
+
+        Return 2 tuples of the form (date, attendance). Each of the tuples can
+        be None if there is no match.
+        """
+        # we should get the first attendance from a list of all the
+        # attendances that starts with the weekday of date_from then wraps
+        # around and ends with the weekday before date_from. the list should
+        # be no longer (in terms of weekdays) than date_to - date_from + 1.
+        all_attendances = self.attendance_ids
+        if not all_attendances:
+            return (None, None)
+        weekday_from = date_from.weekday()
+        weekday_to = date_to.weekday()
+        all_attendances = all_attendances.filtered(
+            lambda r: int(r.dayofweek) >= weekday_from
+        ) + all_attendances.filtered(lambda r: int(r.dayofweek) < weekday_from)
```

Comment:
I have to assume that you meant this:

```suggestion
        all_attendances = all_attendances.filtered(
            lambda r: int(r.dayofweek) >= weekday_from
        ) + all_attendances.filtered(lambda r: int(r.dayofweek) <= weekday_to)
```

(`<= weekday_to` at the end of the last line).


To the best of my reading abilities, the current statement simplifies to `all_attendances = all_attendances`.

Reply:
I was wrong, and this needs a comment. You're re-ordering the recordset so that it starts on `weekday_from`. I sometimes forget that Odoo sets are ordered…

Reply:
This no longer needs to a comment because the methods were split.